runtime/binding/python/CMakeLists.txt

```diff
@@ -22,17 +22,40 @@ if(NOT MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -pthread -fPIC")
 endif()
 
-# libtorch: here we use libtorch from python torch
-execute_process(
-  COMMAND python -c "import os; import torch; print(os.path.dirname(torch.__file__))"
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  OUTPUT_VARIABLE TORCH_DIR
+# # libtorch: here we use libtorch from python torch
+# execute_process(
+#   COMMAND python -c "import os; import torch; print(os.path.dirname(torch.__file__))"
+#   OUTPUT_STRIP_TRAILING_WHITESPACE
+#   OUTPUT_VARIABLE TORCH_DIR
+# )
+# list(APPEND CMAKE_PREFIX_PATH "${TORCH_DIR}")
+# find_package(Torch REQUIRED PATHS ${TORCH_DIR} NO_DEFAULT_PATH)
+# set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS}")
+
+# third_party: libtorch use FetchContent_Declare to download, and
+# use find_package to find since libtorch is not a standard cmake project
+set(PYTORCH_VERSION "1.10.0")
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
+    set(LIBTORCH_URL "https://download.pytorch.org/libtorch/cpu/libtorch-win-shared-with-deps-${PYTORCH_VERSION}%2Bcpu.zip")
+    set(URL_HASH "SHA256=d7043b7d7bdb5463e5027c896ac21b83257c32c533427d4d0d7b251548db8f4b")
+    set(CMAKE_BUILD_TYPE "Release")
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+    set(LIBTORCH_URL "https://download.pytorch.org/libtorch/cpu/libtorch-cxx11-abi-shared-with-deps-${PYTORCH_VERSION}%2Bcpu.zip")
+    set(URL_HASH "SHA256=6d7be1073d1bd76f6563572b2aa5548ad51d5bc241d6895e3181b7dc25554426")
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+    set(LIBTORCH_URL "https://download.pytorch.org/libtorch/cpu/libtorch-macos-${PYTORCH_VERSION}.zip")
+    set(URL_HASH "SHA256=07cac2c36c34f13065cb9559ad5270109ecbb468252fb0aeccfd89322322a2b5")
+else()
+    message(FATAL_ERROR "Unsupported CMake System Name '${CMAKE_SYSTEM_NAME}' (expected 'Windows', 'Linux' or 'Darwin')")
+endif()
+FetchContent_Declare(libtorch
+  URL      ${LIBTORCH_URL}
+  URL_HASH ${URL_HASH}
 )
-list(APPEND CMAKE_PREFIX_PATH "${TORCH_DIR}")
-find_package(Torch REQUIRED PATHS ${TORCH_DIR} NO_DEFAULT_PATH)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS}")
-message(STATUS ${TORCH_CXX_FLAGS})
-message(STATUS "${CMAKE_CXX_FLAGS}")
```

Comment:
Why rollback to libtorch？

Reply:
The feature is still under developing, i'm trying to figure out the right way to make it work.